app/Console/Commands/WaitingLists/CheckMembersMeetActivityRules.php

```diff
@@ -0,0 +1,160 @@
+use App\Models\Mship\Account;
+use App\Models\NetworkData\Atc;
+use App\Models\Training\WaitingList;
+use App\Notifications\Training\WaitingListRemovalAdded;
+use App\Notifications\Training\WaitingListRemovalCancelled;
+use App\Notifications\Training\WaitingListRemovalCompleted;
+use App\Notifications\Training\WaitingListRemovalReminder;
+use Carbon\Carbon;
+
+class CheckMembersMeetActivityRules extends Command
+{
+
+    protected $minutesRequired = 720; // 12 hours is represented as 720 minutes
+
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'waitinglists:hourchecker';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Checks ATC hour totals for members who are on an ATC waiting list and marks inactive members for automatic removal from the list.';
+
+    /**
+     * A local cache of account activity to prevent duplicate calculations of a members hours
+     * 
+     * @var array
+     */
+    protected $accountActivityCache = [];
+
+    /**
+     * Executes all necessary console commands.
+     */
+    public function handle()
+    {
+        foreach($this->retrieveHourEnforcedWaitingLists() as $waitingList){
+            $this->identifyNewNonEligibleWaitingListAccounts($waitingList);
+            $this->processWaitingListAccountsMarkedForRemoval($waitingList);
+       }
+       
+    }
+
+    /**
+     * Retrieves training waiting lists which enforce hour requirements
+     */
+    protected function retrieveHourEnforcedWaitingLists()
+    {
+        return WaitingList::where('enforce_hour_requirement', '=', 'yes')->get();
+    }
+
+    /**
+     * Identifies accounts on enforced waiting lists who have fallen below the hour requirement
+     * 
+     * @param WaitingList $waitingList
+     */
+    protected function identifyNewNonEligibleWaitingListAccounts(WaitingList $waitingList)
+    {
+        // We only want to find Active waiting list accounts which are not already marked for removal
+        $waitingListAccounts                                        = $waitingList->accounts
+            ->filter(function($account){
+                return (
+                    $account->pivot->current_status->name == 'Active' &&
+                    (
+                        is_null($account->pivot->pending_removal->removal_date ?? null) ||
+                        $account->pivot->pending_removal->status != 'Pending'
+                    )
+                );
+            });
+        
+        foreach($waitingListAccounts as $account){
+            // Calculate the total activity within the last 3 months
+            if(isset($this->accountActivityCache[$account->id])){
+                $activeMinutes                                      = $this->accountActivityCache[$account->id];
+            } else {
+                $activeMinutes                                      = Atc::where('account_id', $account->id)
```

Comment:
Regardless of my recommendation on removing the local cache, this logic is duplicated from https://github.com/VATSIM-UK/core/blob/c3ac806defa080dd5a67be737aae3c24788148e1/app/Models/Training/WaitingList/WaitingListAccount.php#L122

Since you are already iterating over instances of `WaitingListAccount` you should just call this method.

Reply:
> Regardless of my recommendation on removing the local cache, this logic is duplicated from https://github.com/VATSIM-UK/core/blob/c3ac806defa080dd5a67be737aae3c24788148e1/app/Models/Training/WaitingList/WaitingListAccount.php#L122
> 
> 
> 
> Since you are already iterating over instances of `WaitingListAccount` you should just call this method. 

I have just realised why I did it this way, and it's because the existing method caches with a 24hour TTL.

That is (in my opinion) an unacceptable amount of caching for this particular function, and if a user meets eligibility after being told that they are on the removal process, then they should be informed that this is acknowledged by the system as soon as it reasonably practical (in this case it will be a maximum of 12hours if the scheduled task runs twice daily)

Reply:
That would mean the email would have a different timespan for the people viewing the waiting lists. This I find to be unacceptable. They need to use the same method of detecting activity, or if someone raises an issue, the human who does the follow-up gets a different view of the system doing the automated checking.

Reply:
Hi @AxonC 

That is fair, however it is worth considering if this caching can be reduced - perhaps halved to 12 hours? 

What would the performance cost be of this?